### ISSUE TITLE
Is this expected behavior? 

### ISSUE THREAD

**Issue by bklingen** [jrowen/rhandsontable#135]
The following code returns a table that keeps on updating ad infinitum. Is this expected behavior? The reason is that I add one row inside the renderRHansontable 
Careful: running this codes generates a table that keeps on getting longer:
```{r}
df0 <- data.frame(Var1=2:4, Var2=9:7)  
    
ui <- fluidPage(
  rHandsontableOutput("hot")
)

server <- function(input, output) {
  output$hot <- renderRHandsontable({
    if(is.null(input$hot)) {
      df <- df0
    } else df <- hot_to_r(input$hot)
    ### add a row:
    df <- rbind(c(1,1), df)
    rhandsontable(df)
  })
}

shinyApp(ui = ui, server = server)
```

**Comment by stla** [comment]
Yes, this is normal, each time the table is rendered, `input$hot` is updated, and then there is a reaction.

**Comment by bklingen** [comment]
I'm not sure. I think this is related to #137 where the code seems to be executed twice.
In the code above, I expect anything inside ``renderRHandsontable({...})`` to be executed once, creating one df, and then rendering it.
To me, the code contains nothing that suggests it is executed repeatedly, i.e., keep on adding rows, but I may not quiet understand how reactivity works between rhandsontable and shiny.

**Comment by stla** [comment]
`output$hot` contains `input$xxx` (with `xxx`=`hot`). Consequently, by the general behaviour of Shiny reactivity (this has nothing to do with `rhandsontable`, `output$hot` is updated each time `input$xxx` changes. 
But here `xxx`=`hot`, and then `input$xxx` changes each time `output$hot` changes (this has to do with `rhandsontable`). 

=> circular.

**Comment by jrowen** [comment]
`rhandsontable` tries to be both a shiny input and output.  It's built with a shiny output binding (via the `htmlwidgets` framework) and uses `Shiny.onInputChange` to mimic input behavior.  This can occasionally lead to unexpected behavior (pass `debug = 2` to the `rhandsontable` and check the JavaScript console for more logging).  There are certainly opportunities to improve the package, and PRs/ideas are very welcome.

**Comment by bklingen** [comment]
Thank you, I think I understand. 
Is it then possible to take the input from input$hot, (i.e., grab the dataframe created by hot_to_r(input$hot)), do something to it (such as adding a row), and then display that modified dataframe as the "new" rhandsontable? I will try to provide a reproducible example, but I think it will have to use reactiveValues.
